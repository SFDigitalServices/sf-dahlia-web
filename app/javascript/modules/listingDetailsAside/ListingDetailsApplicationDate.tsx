import React from "react"
import { RailsListing } from "../listings/SharedHelpers"
import { ApplicationStatus, ApplicationStatusType, Icon, t } from "@bloom-housing/ui-components"
import { localizedFormat } from "../../util/languageUtil"
import dayjs from "dayjs"
import { LISTING_TYPE_FIRST_COME_FIRST_SERVED } from "../constants"
import { useFeatureFlag } from "../../hooks/useFeatureFlag"
import { isFcfsSalesListing, isOpen } from "../../util/listingUtil"
import bloomTheme from "../../../../tailwind.config"
import { Message } from "@bloom-housing/ui-seeds"

export interface ListingDetailsApplicationDateProps {
  listing: RailsListing
}

export const formatTime = (time: string) => {
  const formattedTime = dayjs(time).format("h:mm")
  const hour = Number(dayjs(time).format("H"))
  // \u00A0 is a non-breaking space
  const suffix = hour >= 12 ? "\u00A0PM" : "\u00A0AM"
  return `${formattedTime}${suffix}`
}

const StatusMessage = ({ isClosed, message }: { isClosed: boolean; message: string }) => {
  return (
    <Message
      fullwidth
      className="justify-start leading-5"
      variant={isClosed ? "alert" : "primary"}
      customIcon={<Icon fill={isClosed ? "red-700" : ""} symbol="clock" size="medium" />}
    >
      {message}
    </Message>
  )
}

const getStatusLottery = (listing: RailsListing) => {
  const isClosed = !isOpen(listing)
  const datetime = listing.Application_Due_Date
  return (
    <StatusMessage
      isClosed={isClosed}
      message={t(
        isClosed
          ? "listingDetails.applicationsClosed.withDateTime"
          : "listingDetails.applicationsDeadline.withDateTime",
        {
          date: localizedFormat(datetime, "LL"),
          time: formatTime(datetime),
        }
      )}
    />
  )
}

const getStatusFcfs = (listing: RailsListing) => {
  const isClosed = !listing.Accepting_Online_Applications
  const datetime = listing.Application_Start_Date_Time
  const showDateTime = new Date(listing.Application_Start_Date_Time) > new Date()

  const getMessage = () => {
    if (isClosed) {
      return t("listingDetails.applicationsClosed")
    } else {
      return showDateTime
        ? t("listingDetails.applicationsOpen.withDateTime", {
            date: localizedFormat(datetime, "LL"),
            time: formatTime(datetime),
          })
        : t("listingDetails.applicationsOpen")
    }
  }

  return <StatusMessage isClosed={isClosed} message={getMessage()} />
}

const ListingDetailsStatus = ({ listing }: { listing: RailsListing }) => {
  return listing.Listing_Type === LISTING_TYPE_FIRST_COME_FIRST_SERVED
    ? getStatusFcfs(listing)
    : getStatusLottery(listing)
}

export const ListingDetailsApplicationDate = ({ listing }: ListingDetailsApplicationDateProps) => {
  const isApplicationOpen = listing && isOpen(listing)

  const { unleashFlag: isSalesFcfsEnabled } = useFeatureFlag("FCFS", false)

  return (
    <div className="w-full mb-8 md:mb-0">
      {isSalesFcfsEnabled && isFcfsSalesListing(listing) ? (
        <ListingDetailsStatus listing={listing} />
      ) : (
        <div className="w-full mb-8 md:mb-0">
          <ApplicationStatus
            className="place-content-center"
            content={t(
              isApplicationOpen
                ? "listingDetails.applicationDeadline.open"
                : "listingDetails.applicationDeadline.closed",
              {
                date: localizedFormat(listing.Application_Due_Date, "ll"),
                time: dayjs(listing.Application_Due_Date).format("h:mm A"),
              }
            )}
            iconColor={!isApplicationOpen && bloomTheme.theme.colors.red["700"]}
            status={isApplicationOpen ? ApplicationStatusType.Open : ApplicationStatusType.Closed}
          />
        </div>
      )}
    </div>
  )
}
