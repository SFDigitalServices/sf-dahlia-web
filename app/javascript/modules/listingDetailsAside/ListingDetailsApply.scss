.fcfs-bmr-how-to-apply {
  padding: var(--seeds-spacer-content) !important;
}

.fcfs-bmr-how-to-apply__list {
  padding-top: var(--seeds-spacer-content) !important;
  padding-bottom: var(--seeds-spacer-content) !important;
}

.fcfs-bmr-how-to-apply {
  h4.text__underline-weighted {
    margin-bottom: 0px;
  }
}
