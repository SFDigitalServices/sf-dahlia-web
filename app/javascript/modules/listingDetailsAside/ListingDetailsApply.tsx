import React, { useState } from "react"
import {
  AppearanceStyleType,
  Button,
  Heading,
  LinkButton,
  MultiLineAddress,
  OrDivider,
  SidebarBlock,
  t,
} from "@bloom-housing/ui-components"
import { RailsListing } from "../listings/SharedHelpers"
import {
  acceptingPaperApplications,
  isHabitatListing,
  isOpen,
  isRental,
  isSale,
  paperApplicationURLs,
} from "../../util/listingUtil"
import { localizedFormat, renderInlineMarkup } from "../../util/languageUtil"

export interface ListingDetailsApplyProps {
  listing: RailsListing
}

export const ListingDetailsApply = ({ listing }: ListingDetailsApplyProps) => {
  const [paperApplicationsOpen, setPaperApplicationsOpen] = useState(false)

  if (!isOpen(listing)) return null

  const isListingRental = isRental(listing)

  const acceptingPaperApps = acceptingPaperApplications(listing)

  const ordinalHeader = (ordinal: number, title: string) => {
    return (
      <Heading priority={4} className={"text-gray-900 text-lg -mt-2 mb-3"}>
        <span className={"text-blue-600 mr-2"}>{ordinal}</span>
        {title}
      </Heading>
    )
  }

  const howToApplyBlock = (
    <SidebarBlock title={t("listings.apply.howToApply")}>
      {!isListingRental && (
        <>
          <p className={"mb-4"}>
            {renderInlineMarkup(
              t("listings.apply.fulfillEligibilityRequirements", {
                url: isHabitatListing(listing)
                  ? "https://habitatgsf.org/amber-drive-info/"
                  : "https://sfmohcd.org/homebuyer-program-eligibility",
              })
            )}
          </p>
          <p className={"mb-4"}>{t("listings.apply.eligibilityRequirementDescription")}</p>
        </>
      )}
      <LinkButton
        styleType={AppearanceStyleType.primary}
        className={"w-full"}
        transition={true}
        href={`listings/${listing.listingID}/apply-welcome/intro`}
      >
        {t("label.applyOnline")}
      </LinkButton>
      {process.env.COVID_UPDATE && (
        <div className={"mt-4"}>
          <Heading priority={4} className={"text-base text-gray-800 font-sans"}>
            {t("listings.apply.covidUpdate")}
          </Heading>
          <div className={"text-gray-700 text-base mt-2"}>
            {t("listings.apply.covidUpdateInfo")}
          </div>
        </div>
      )}
      {acceptingPaperApps && (
        <>
          <OrDivider bgColor={"white"} />
          {ordinalHeader(1, t("listings.apply.getAPaperApplication"))}
          {t("listings.apply.paperApplicationsMustBeMailed")}
          <Button
            className={"w-full mt-4"}
            transition={true}
            icon={"arrowDown"}
            iconPlacement={"right"}
            onClick={() => setPaperApplicationsOpen(!paperApplicationsOpen)}
          >
            {t("label.downloadApplication")}
          </Button>
          {paperApplicationsOpen && acceptingPaperApplications && (
            <div className={"flex w-full items-center justify-center flex-col"}>
              {paperApplicationURLs(isListingRental).map((app) => {
                return (
                  <span className={"pt-4"}>
                    <LinkButton
                      href={app.fileURL}
                      unstyled
                      className={"m-0 no-underline"}
                      newTab={true}
                    >
                      {app.languageString}
                    </LinkButton>
                  </span>
                )
              })}
            </div>
          )}
        </>
      )}
    </SidebarBlock>
  )

  const submitPaperApplicationBlocks = (
    <>
      <SidebarBlock className={"bg-blue-200"}>
        {ordinalHeader(2, t("listings.apply.submitAPaperApplication"))}
        {t("listings.apply.includeAnEnvelope")}
      </SidebarBlock>
      <SidebarBlock
        className={"bg-blue-200"}
        styleType={"capsWeighted"}
        title={t("listings.apply.sendByUsMail")}
      >
        <div className={"mb-2 text-gray-900 text-base"}>
          <MultiLineAddress
            address={{
              placeName: listing.Application_Organization,
              street: listing.Application_Street_Address,
              city: listing.Application_City,
              zipCode: listing.Application_Postal_Code,
              state: listing.Application_State,
            }}
          />
        </div>
        {t("listings.apply.applicationsMustBeReceivedByDeadline")}
      </SidebarBlock>
      <SidebarBlock className={"bg-blue-200"}>
        {t("listings.doNotApplyOneAndPaperOrMultiple")}
      </SidebarBlock>
    </>
  )

  const needHelpBlock = (
    <SidebarBlock title={t("listings.apply.needHelp")}>
      {isListingRental && (
        <div className={"mb-4"}>{t("listings.apply.visitAHousingCounselor")}</div>
      )}
      <LinkButton
        transition={true}
        newTab={true}
        href={
          !isListingRental
            ? "https://www.homeownershipsf.org/application-assistance-for-homebuyers/"
            : "/housing-counselors"
        }
        className={"w-full"}
      >
        {isListingRental
          ? t("housingCounselor.findAHousingCounselor")
          : t("listings.apply.visitHomeownershipSf")}
      </LinkButton>
    </SidebarBlock>
  )

  const expectedMoveInDateBlock = (
    <SidebarBlock title={t("listings.expectedMoveinDate")}>
      {localizedFormat(listing.Expected_Move_in_Date, "MMMM YYYY")}
    </SidebarBlock>
  )

  return (
    <>
      {howToApplyBlock}
      {acceptingPaperApps && submitPaperApplicationBlocks}
      {needHelpBlock}
      {isSale(listing) && listing.Expected_Move_in_Date && expectedMoveInDateBlock}
    </>
  )
}
