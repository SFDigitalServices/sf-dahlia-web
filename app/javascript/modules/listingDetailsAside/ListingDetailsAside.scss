.fcfs-aside {
    --card-border-radius: rounded-none;
    --card-background-color: var(--bloom-color-gray-200);
    --card-footer-background-color: var(--bloom-color-gray-200);
    --card-border-width: border-b-1;

    .aside-card-header {
        font-family: var(--seeds-font-alt-sans);
        font-size: var(--seeds-font-size-lg);
        font-weight: var(-seeds-font-weight-semibold);
        line-height: var(--seeds-line-height-6);
    }

    .aside-card-section {
        font-family: var(--seeds-font-sans);
        font-size: var(--seeds-font-size-sm);
        font-weight: var(--seeds-font-weight-regular);
        line-height: var(--seeds-line-height-5);
    }
}
