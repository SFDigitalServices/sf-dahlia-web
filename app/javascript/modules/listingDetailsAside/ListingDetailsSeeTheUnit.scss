.see-the-unit {
  padding: var(--seeds-spacer-content) !important;

  .see-the-unit__subsection:not(:last-of-type) {
    padding-bottom: var(--seeds-spacer-content);
  }

  .see-the-unit__heading {
    h4.text__underline-weighted {
      margin-bottom: 0px !important;
    }
    padding-bottom: var(--seeds-spacer-content);
  }
}
