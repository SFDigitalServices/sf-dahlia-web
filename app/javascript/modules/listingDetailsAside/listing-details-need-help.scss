.listing-details-need-help {
  padding: var(--seeds-spacer-content) !important;
  border-color: var(--bloom-color-gray-400) !important;
}
