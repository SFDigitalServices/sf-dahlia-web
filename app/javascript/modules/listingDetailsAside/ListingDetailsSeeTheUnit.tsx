import React from "react"
import { RailsListing } from "../../modules/listings/SharedHelpers"
import "./ListingDetailsSeeTheUnit.scss"
import { t, Heading, Icon, IconFillColors } from "@bloom-housing/ui-components"
import { Heading as HeadingSeeds } from "@bloom-housing/ui-seeds"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { faEnvelope } from "@fortawesome/free-solid-svg-icons"
import { ListingDetailsOpenHouses } from "./ListingDetailsOpenHouses"

export interface SeeTheUnitProps {
  listing: RailsListing
}

export interface SubsectionProps {
  title?: string
  children: React.ReactNode
}

const SeeTheUnitSubsection = ({ title, children }: SubsectionProps) => {
  return (
    <div className="see-the-unit__subsection">
      {title && (
        <HeadingSeeds size="md" className="pb-3">
          {title}
        </HeadingSeeds>
      )}
      <div>{children}</div>
    </div>
  )
}

export const ListingDetailsSeeTheUnit = ({ listing }: SeeTheUnitProps) => {
  return (
    <section className="aside-block see-the-unit">
      <div className="see-the-unit__heading">
        <Heading priority={4} styleType="underlineWeighted">
          {t("label.seeTheUnit")}
        </Heading>
      </div>
      <SeeTheUnitSubsection title={t("label.openHouses.seeTheUnit")}>
        {listing.Open_Houses?.length ? (
          <ListingDetailsOpenHouses listing={listing} sectionHeader={false} />
        ) : null}
      </SeeTheUnitSubsection>
      {listing.Multiple_Listing_Service_URL && (
        <SeeTheUnitSubsection title={t("seeTheUnit.seeDetailsOnline")}>
          <p>
            <a href={listing.Multiple_Listing_Service_URL} target="_blank">
              {t("listings.process.seeTheUnitOnMls")}
            </a>
          </p>
        </SeeTheUnitSubsection>
      )}
      <SeeTheUnitSubsection title={t("seeTheUnit.makeAnAppointment")}>
        <div className="see-the-unit__subsection-content">
          <p>{t("seeTheUnit.requestATour")}</p>
        </div>
      </SeeTheUnitSubsection>
      <SeeTheUnitSubsection>
        <p>{listing.Leasing_Agent_Name}</p>
        <p className="text-gray-700 text-sm">{listing.Leasing_Agent_Title}</p>
        <div className="pt-2">
          <p className="pb-1">
            <a
              href={
                listing.Leasing_Agent_Phone
                  ? `tel:${listing.Leasing_Agent_Phone.replace(/[-()]/g, "")}`
                  : undefined
              }
            >
              <Icon symbol="phone" size="medium" fill={IconFillColors.primary} className={"pr-2"} />
              {listing.Leasing_Agent_Phone
                ? t("listings.call", { phoneNumber: listing.Leasing_Agent_Phone })
                : undefined}
            </a>
          </p>
          <p className="pb-3">
            <a href={`mailto:${listing.Leasing_Agent_Email}`}>
              <span className="pr-2">
                <FontAwesomeIcon icon={faEnvelope} />
              </span>
              {t("label.emailAddress")}
            </a>
          </p>
        </div>
        <HeadingSeeds size="md">{t("contactAgent.officeHours.seeTheUnit")}</HeadingSeeds>
        <div className="see-the-unit__subsection-content">
          <p>{listing.Office_Hours}</p>
        </div>
      </SeeTheUnitSubsection>
    </section>
  )
}
