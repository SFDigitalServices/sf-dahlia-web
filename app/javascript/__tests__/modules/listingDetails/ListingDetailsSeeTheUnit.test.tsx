import React from "react"
import { render } from "@testing-library/react"
import { openSaleListing } from "../../data/RailsSaleListing/listing-sale-open"
import { saleEducatorListing } from "../../data/RailsSaleListing/listing-sale-educator"
import { onlineDetailsSaleListing } from "../../data/RailsSaleListing/listing-sale-online-details"
import { mlsOnlineDetailsSaleListing } from "../../data/RailsSaleListing/listing-sale-mls-online-details"
import { ListingDetailsSeeTheUnit } from "../../../modules/listingDetailsAside/ListingDetailsSeeTheUnit"

describe("ListingDetailsSeeTheUnit", () => {
  it("does not render SeeDetailsOnline when MLS and Listing_Online_Details are not set", () => {
    // saleEducatorListing does not have Multiple_Listing_Service_URL and Listing_Online_Details
    const { asFragment } = render(<ListingDetailsSeeTheUnit listing={saleEducatorListing} />)

    // this snapshot should not include SeeDetailsOnline
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders SeeDetailsOnline when only MLS is set", () => {
    // openSaleListing has Multiple_Listing_Service_URL defined
    const { asFragment } = render(<ListingDetailsSeeTheUnit listing={openSaleListing} />)

    // this snapshot should include SeeDetailsOnline with only the MLS link
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders SeeDetailsOnline when only Listing_Online_Details is set", () => {
    // onlineDetailsSaleListing has Listing_Online_Details defined
    const { asFragment } = render(<ListingDetailsSeeTheUnit listing={onlineDetailsSaleListing} />)

    // this snapshot should include SeeDetailsOnline with only the Listing_Online_Details link
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders SeeDetailsOnline when both MLS and Listing_Online_Details are set", () => {
    // mlsOnlineDetailsSaleListing has both MLS and Listing_Online_Details defined
    const { asFragment } = render(
      <ListingDetailsSeeTheUnit listing={mlsOnlineDetailsSaleListing} />
    )

    // this snapshot should include SeeDetailsOnline with both MLS and Listing_Online_Details links
    expect(asFragment()).toMatchSnapshot()
  })
})
