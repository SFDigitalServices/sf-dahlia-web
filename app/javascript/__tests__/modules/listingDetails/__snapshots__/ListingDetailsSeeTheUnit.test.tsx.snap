// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetailsSeeTheUnit does not render SeeDetailsOnline when MLS and Listing_Online_Details are not set 1`] = `
<DocumentFragment>
  <section
    class="aside-block see-the-unit translate"
  >
    <div
      class="see-the-unit__heading"
    >
      <h2
        class="text__underline-weighted"
      >
        See the Unit
      </h2>
    </div>
    <div
      class="see-the-unit__subsection"
    >
      <h3
        class="text-heading-md pb-3"
      >
        Make an appointment
      </h3>
      <div>
        <p
          class="text-sm pb-3"
        >
          Call, email, or text to request a tour.
        </p>
        <p />
        <p
          class="text-gray-700 text-sm"
        />
        <div
          class="pt-2"
        >
          <p
            class="pb-1"
          >
            <a>
              <span
                class="ui-icon ui-medium pr-2"
              >
                <svg
                  fill="#0077DA"
                  viewBox="0 0 32 32"
                >
                  <path
                    d="M22 20c-2 2-2 4-4 4s-4-2-6-4-4-4-4-6 2-2 4-4-4-8-6-8-6 6-6 6c0 4 4.109 12.109 8 16s12 8 16 8c0 0 6-4 6-6s-6-8-8-6z"
                  />
                </svg>
              </span>
            </a>
          </p>
          <p
            class="pb-3"
          >
            <a
              href="mailto:undefined"
            >
              <span
                class="pr-2"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-envelope "
                  data-icon="envelope"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48L48 64zM0 176L0 384c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-208L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              Email
            </a>
          </p>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ListingDetailsSeeTheUnit renders SeeDetailsOnline when both MLS and Listing_Online_Details are set 1`] = `
<DocumentFragment>
  <section
    class="aside-block see-the-unit translate"
  >
    <div
      class="see-the-unit__heading"
    >
      <h2
        class="text__underline-weighted"
      >
        See the Unit
      </h2>
    </div>
    <div
      class="see-the-unit__subsection"
    >
      <h3
        class="text-heading-md pb-3"
      >
        Open houses
      </h3>
      <div>
        <div
          class="open-houses"
        >
          <section
            class=""
          >
            <div
              class="pb-3"
            >
              <p
                class="text text-gray-800 pb-2 flex justify-between items-center"
              >
                <span
                  class="inline-block text-sm uppercase normal-case"
                >
                  December 2, 2019
                </span>
                <span
                  class="inline-block text-xs font-bold ml-5 font-alt-sans"
                >
                  2:00 PM - 6:00 PM
                </span>
              </p>
            </div>
            <div
              class="pb-3"
            >
              <p
                class="text text-gray-800 pb-2 flex justify-between items-center"
              >
                <span
                  class="inline-block text-sm uppercase normal-case"
                >
                  December 6, 2019
                </span>
                <span
                  class="inline-block text-xs font-bold ml-5 font-alt-sans"
                >
                  10:00 AM - 12:00 PM
                </span>
              </p>
              <div
                class="text-sm text-gray-700 pb-3"
              >
                <div
                  class="flex flex-col"
                >
                  <span
                    class="links-space translate"
                  >
                    <span>
                      SFPL
                    </span>
                  </span>
                  <span>
                    1 Library Way, San Francisco
                  </span>
                </div>
              </div>
            </div>
            <div
              class="false"
            >
              <p
                class="text text-gray-800 pb-2 flex justify-between items-center"
              >
                <span
                  class="inline-block text-sm uppercase normal-case"
                >
                  December 7, 2019
                </span>
              </p>
            </div>
          </section>
        </div>
      </div>
    </div>
    <div
      class="see-the-unit__subsection"
    >
      <h3
        class="text-heading-md pb-3"
      >
        See details online
      </h3>
      <div>
        <div
          class="flex-row space-y-1"
        >
          <div>
            <a
              class="seeds-link no-underline"
              href="https://thefrontsteps.com/featured-listings/sfar/mls-481663-2820_scott_street_san_francisco_ca_94123/"
            >
              See the unit on MLS
            </a>
          </div>
          <div>
            <a
              href="https://www.google.com/"
              target="_blank"
            >
              Online Detail
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="see-the-unit__subsection"
    >
      <h3
        class="text-heading-md pb-3"
      >
        Make an appointment
      </h3>
      <div>
        <p
          class="text-sm pb-3"
        >
          Call, email, or text to request a tour.
        </p>
        <p>
          Matthew Burns
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          Leasing Agent
        </p>
        <div
          class="pt-2"
        >
          <p
            class="pb-1"
          >
            <a
              href="tel:415 5552323"
            >
              <span
                class="ui-icon ui-medium pr-2"
              >
                <svg
                  fill="#0077DA"
                  viewBox="0 0 32 32"
                >
                  <path
                    d="M22 20c-2 2-2 4-4 4s-4-2-6-4-4-4-4-6 2-2 4-4-4-8-6-8-6 6-6 6c0 4 4.109 12.109 8 16s12 8 16 8c0 0 6-4 6-6s-6-8-8-6z"
                  />
                </svg>
              </span>
              Call (415) 555-2323
            </a>
          </p>
          <p
            class="pb-3"
          >
            <a
              href="mailto:ae@email.com"
            >
              <span
                class="pr-2"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-envelope "
                  data-icon="envelope"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48L48 64zM0 176L0 384c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-208L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              Email
            </a>
          </p>
        </div>
        <h3
          class="text-heading-sm pb-1"
        >
          Office hours
        </h3>
        <p
          class="text-sm"
        >
          <span>
            8-4 Mon-Thurs

            <br />
            9-4 Fri

            <br />
            10-2 Weekends
          </span>
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ListingDetailsSeeTheUnit renders SeeDetailsOnline when only Listing_Online_Details is set 1`] = `
<DocumentFragment>
  <section
    class="aside-block see-the-unit translate"
  >
    <div
      class="see-the-unit__heading"
    >
      <h2
        class="text__underline-weighted"
      >
        See the Unit
      </h2>
    </div>
    <div
      class="see-the-unit__subsection"
    >
      <h3
        class="text-heading-md pb-3"
      >
        Open houses
      </h3>
      <div>
        <div
          class="open-houses"
        >
          <section
            class=""
          >
            <div
              class="pb-3"
            >
              <p
                class="text text-gray-800 pb-2 flex justify-between items-center"
              >
                <span
                  class="inline-block text-sm uppercase normal-case"
                >
                  December 2, 2019
                </span>
                <span
                  class="inline-block text-xs font-bold ml-5 font-alt-sans"
                >
                  2:00 PM - 6:00 PM
                </span>
              </p>
            </div>
            <div
              class="pb-3"
            >
              <p
                class="text text-gray-800 pb-2 flex justify-between items-center"
              >
                <span
                  class="inline-block text-sm uppercase normal-case"
                >
                  December 6, 2019
                </span>
                <span
                  class="inline-block text-xs font-bold ml-5 font-alt-sans"
                >
                  10:00 AM - 12:00 PM
                </span>
              </p>
              <div
                class="text-sm text-gray-700 pb-3"
              >
                <div
                  class="flex flex-col"
                >
                  <span
                    class="links-space translate"
                  >
                    <span>
                      SFPL
                    </span>
                  </span>
                  <span>
                    1 Library Way, San Francisco
                  </span>
                </div>
              </div>
            </div>
            <div
              class="false"
            >
              <p
                class="text text-gray-800 pb-2 flex justify-between items-center"
              >
                <span
                  class="inline-block text-sm uppercase normal-case"
                >
                  December 7, 2019
                </span>
              </p>
            </div>
          </section>
        </div>
      </div>
    </div>
    <div
      class="see-the-unit__subsection"
    >
      <h3
        class="text-heading-md pb-3"
      >
        See details online
      </h3>
      <div>
        <div
          class="flex-row space-y-1"
        >
          <div>
            <a
              href="https://www.google.com/"
              target="_blank"
            >
              Online Detail
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="see-the-unit__subsection"
    >
      <h3
        class="text-heading-md pb-3"
      >
        Make an appointment
      </h3>
      <div>
        <p
          class="text-sm pb-3"
        >
          Call, email, or text to request a tour.
        </p>
        <p>
          Matthew Burns
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          Leasing Agent
        </p>
        <div
          class="pt-2"
        >
          <p
            class="pb-1"
          >
            <a
              href="tel:415 5552323"
            >
              <span
                class="ui-icon ui-medium pr-2"
              >
                <svg
                  fill="#0077DA"
                  viewBox="0 0 32 32"
                >
                  <path
                    d="M22 20c-2 2-2 4-4 4s-4-2-6-4-4-4-4-6 2-2 4-4-4-8-6-8-6 6-6 6c0 4 4.109 12.109 8 16s12 8 16 8c0 0 6-4 6-6s-6-8-8-6z"
                  />
                </svg>
              </span>
              Call (415) 555-2323
            </a>
          </p>
          <p
            class="pb-3"
          >
            <a
              href="mailto:ae@email.com"
            >
              <span
                class="pr-2"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-envelope "
                  data-icon="envelope"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48L48 64zM0 176L0 384c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-208L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              Email
            </a>
          </p>
        </div>
        <h3
          class="text-heading-sm pb-1"
        >
          Office hours
        </h3>
        <p
          class="text-sm"
        >
          <span>
            8-4 Mon-Thurs

            <br />
            9-4 Fri

            <br />
            10-2 Weekends
          </span>
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ListingDetailsSeeTheUnit renders SeeDetailsOnline when only MLS is set 1`] = `
<DocumentFragment>
  <section
    class="aside-block see-the-unit translate"
  >
    <div
      class="see-the-unit__heading"
    >
      <h2
        class="text__underline-weighted"
      >
        See the Unit
      </h2>
    </div>
    <div
      class="see-the-unit__subsection"
    >
      <h3
        class="text-heading-md pb-3"
      >
        Open houses
      </h3>
      <div>
        <div
          class="open-houses"
        >
          <section
            class=""
          >
            <div
              class="pb-3"
            >
              <p
                class="text text-gray-800 pb-2 flex justify-between items-center"
              >
                <span
                  class="inline-block text-sm uppercase normal-case"
                >
                  December 2, 2019
                </span>
                <span
                  class="inline-block text-xs font-bold ml-5 font-alt-sans"
                >
                  2:00 PM - 6:00 PM
                </span>
              </p>
            </div>
            <div
              class="pb-3"
            >
              <p
                class="text text-gray-800 pb-2 flex justify-between items-center"
              >
                <span
                  class="inline-block text-sm uppercase normal-case"
                >
                  December 6, 2019
                </span>
                <span
                  class="inline-block text-xs font-bold ml-5 font-alt-sans"
                >
                  10:00 AM - 12:00 PM
                </span>
              </p>
              <div
                class="text-sm text-gray-700 pb-3"
              >
                <div
                  class="flex flex-col"
                >
                  <span
                    class="links-space translate"
                  >
                    <span>
                      SFPL
                    </span>
                  </span>
                  <span>
                    1 Library Way, San Francisco
                  </span>
                </div>
              </div>
            </div>
            <div
              class="false"
            >
              <p
                class="text text-gray-800 pb-2 flex justify-between items-center"
              >
                <span
                  class="inline-block text-sm uppercase normal-case"
                >
                  December 7, 2019
                </span>
              </p>
            </div>
          </section>
        </div>
      </div>
    </div>
    <div
      class="see-the-unit__subsection"
    >
      <h3
        class="text-heading-md pb-3"
      >
        See details online
      </h3>
      <div>
        <div
          class="flex-row space-y-1"
        >
          <div>
            <a
              class="seeds-link no-underline"
              href="https://thefrontsteps.com/featured-listings/sfar/mls-481663-2820_scott_street_san_francisco_ca_94123/"
            >
              See the unit on MLS
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="see-the-unit__subsection"
    >
      <h3
        class="text-heading-md pb-3"
      >
        Make an appointment
      </h3>
      <div>
        <p
          class="text-sm pb-3"
        >
          Call, email, or text to request a tour.
        </p>
        <p>
          Matthew Burns
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          Leasing Agent
        </p>
        <div
          class="pt-2"
        >
          <p
            class="pb-1"
          >
            <a
              href="tel:415 5552323"
            >
              <span
                class="ui-icon ui-medium pr-2"
              >
                <svg
                  fill="#0077DA"
                  viewBox="0 0 32 32"
                >
                  <path
                    d="M22 20c-2 2-2 4-4 4s-4-2-6-4-4-4-4-6 2-2 4-4-4-8-6-8-6 6-6 6c0 4 4.109 12.109 8 16s12 8 16 8c0 0 6-4 6-6s-6-8-8-6z"
                  />
                </svg>
              </span>
              Call (415) 555-2323
            </a>
          </p>
          <p
            class="pb-3"
          >
            <a
              href="mailto:ae@email.com"
            >
              <span
                class="pr-2"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-envelope "
                  data-icon="envelope"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48L48 64zM0 176L0 384c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-208L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              Email
            </a>
          </p>
        </div>
        <h3
          class="text-heading-sm pb-1"
        >
          Office hours
        </h3>
        <p
          class="text-sm"
        >
          <span>
            8-4 Mon-Thurs

            <br />
            9-4 Fri

            <br />
            10-2 Weekends
          </span>
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
